### PR TITLE
RDCC-5443 - Staff : Create a new API for basic search

### PR DESCRIPTION
**Pull request by ishratt**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDCC-5443

### Change description ###
Created new api for searching staff user by name

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/StaffRefFunctionalTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            .andReturn();
+        fetchResponse.then()
+            .assertThat()
+            .statusCode(200);
+
+        List<SearchStaffUserResponse> searchStaffUserResponse = Arrays.asList(
+            fetchResponse.getBody().as(SearchStaffUserResponse[].class));
+        assertThat(searchStaffUserResponse).isNotNull().hasSize(1);
+        assertThat(searchStaffUserResponse.get(0).getFirstName()).contains(SEARCH_STRING);
+        assertThat(searchStaffUserResponse.get(0).getLastName()).contains(SEARCH_STRING);
```
Comment (rajkatla-hmcts) [inline src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/StaffRefFunctionalTest.java:69]: 1. can you check for total records count from the response header
2. add another test case for non matching records with search string and check for total records count from the response header
3. add another test case for non staff admin user
4. add another test case with only with page number and check for total records count from the response header
5. add another test case with only size and check for total records count from the response header
6. add another test case with less than 3 chars of search string
7. add another test case with invalid page
8. add another test case with invalid size

File: src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/StaffRefIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+class StaffRefIntegrationTest extends AuthorizationEnabledIntegrationTest {
+
+    @BeforeEach
+    public void setUpClient() {
+        super.setUpClient();
+    }
+
+    @Test
+    void returns_200_when_users_retrieved_successfully() throws JsonProcessingException {
+        //TODO:
```
Comment (rajkatla-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/StaffRefIntegrationTest.java:17]: 1. can you complete the test case and check for total records count from the response header
2. add another test case for non matching records with search string and check for total records count from the response header
3. add another test case for non staff admin user
4. add another test case with only with page number and check for total records count from the response header
5. add another test case with only size and check for total records count from the response header
6. add another test case for search string less than 3 chars
7. add another test case for null, empty and all possible special chars search string
8. add another test case with invalid page
9. add another test case with invalid size

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/domain/Skill.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    @CreationTimestamp
+    @Column(name = "created_date")
+    private LocalDateTime createdDate;
+
+    @UpdateTimestamp
+    @Column(name = "last_update")
+    private LocalDateTime lastUpdate;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    //@JoinColumn(name = "skill_id", referencedColumnName = "skill_id",
```
Comment (rajkatla-hmcts) [inline src/main/java/uk/gov/hmcts/reform/cwrdapi/domain/Skill.java:56]: remove commented code

**Review by rajkatla-hmcts — CHANGES_REQUESTED** [review]
